src/Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing.Tests/ConventionBasedDecoratorRegistrationGatewayExtensionsTests.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using AutoFixture;
+using AutoFixture.Xunit2;
+using FluentAssertions;
+using Functional.CQS.AOP.CommonTestInfrastructure;
+using Functional.CQS.AOP.CommonTestInfrastructure.MetricsCapturing;
+using Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing.Configuration;
+using SimpleInjector;
+using Xunit;
+
+namespace Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing.Tests
+{
+	public class ConventionBasedDecoratorRegistrationGatewayExtensionsTests
+	{
+		[Theory, SingletonRegistrationWithNoDecorationsEnabledArrangement]
+		public void ShouldNotApplyDecoratorIfConfigurationHasAllDecorationsDisabled_SingletonRegistration(Container container) => ShouldNotApplyDecoratorIfConfigurationHasAllDecorationsDisabled(container);
+		
+		[Theory, TransientRegistrationWithNoDecorationsEnabledArrangement]
+		public void ShouldNotApplyDecoratorIfConfigurationHasAllDecorationsDisabled_TransientRegistration(Container container) => ShouldNotApplyDecoratorIfConfigurationHasAllDecorationsDisabled(container);
+		
+		private static void ShouldNotApplyDecoratorIfConfigurationHasAllDecorationsDisabled(Container container)
```

Comment:
Out of curiosity, is having separate test methods for the `SingletonRegistration` and `TransientRegistration` arrangements  done here because it would otherwise be difficult to identify which arrangement failed in a unit test runner?

Reply:
Yes.  For some reason, the test runner on my machine will show the results of _one_ arrangement if multiple arrangements are applied to one test method.  I do not recall this being a problem, but it's possible that the test runner has always behaved this way.

In any case, it makes distinguishing between the different cases in build logs much easier because the method name includes some arrangement information.